src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,139 @@
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private static final Map<Class<?>, Class<?>> interfacesImplementations;
+    private static final Map<Class<?>, Object> instances = new HashMap<>();
+    private static final String SERVICE_PACKAGE = "mate.academy.service";
+    private static final String SERVICE_PACKAGE_IMPL = "mate.academy.service.impl";
+    private static final String CLASS_INDICATOR = ".class";
+
+    static {
+        interfacesImplementations = interfacesImplementationsConfiguration();
+    }
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        Field[] fields = clazz.getDeclaredFields();
+        for (Field field : fields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                clazzImplementationInstance = createNewInstance(clazz);
+                field.setAccessible(true);
+                try {
+                    field.set(clazzImplementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can't initialize field value. Class: "
+                            + clazz.getName() + ", field: " + field.getName());
+                }
+            }
+        }
+        if (clazzImplementationInstance == null) {
+            clazzImplementationInstance = createNewInstance(clazz);
+        }
+        return clazzImplementationInstance;
+    }
+
+    private static Object createNewInstance(Class<?> clazz) {
+        if (instances.containsKey(clazz)) {
+            return instances.get(clazz);
+        }
+        try {
+            Constructor<?> constructor = clazz.getConstructor();
+            Object instance = constructor.newInstance();
+            instances.put(clazz, instance);
+            return instance;
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException("Can't create new instance of class - " + clazz.getName());
+        }
+    }
+
+    private static Class<?> findImplementation(Class<?> interfaceClazz) {
+        if (interfaceClazz.isInterface()) {
+            if (!interfacesImplementations.containsKey(interfaceClazz)) {
+                throw new RuntimeException("Can`t find interface " + interfaceClazz.getName());
+            }
+            Class<?> implementation = interfacesImplementations.get(interfaceClazz);
+            if (implementation == null) {
+                throw new RuntimeException("Interface " + interfaceClazz.getName()
+                        + " has no implementations or they doesn't annotated by @Component");
+            }
+            return implementation;
+        }
+        componentCheck(interfaceClazz);
```

Comment:
You should decide how to check exceptions: in separate methods or in this method.